### DISCUSSION TITLE
How can I call the hybrid_method of a particular record on all records?

### DISCUSSION THREAD

**Discussion by DUOLabs333** [sqlalchemy/sqlalchemy#10739] [Usage Questions]
I have tables/classes `Table1` and `Table2`. I have a record/object of type `Table1` `foo`. `foo` has a hybrid_method `bar(baz)`. I want to select all objects `tab` of type `Table2` such that `foo.bar(tab.prop)` is True. How would I do that?

As far as I know, instance-specific methods can't operate on columns, though maybe I'm missing some incantation. 

I tried using `func`, but it doesn't seem to work.

Some example code so it's clearer what I'm talking about:
```python
class Table1(BaseTable):
  @hybrid_method
  def bar(self, baz):
    return ...

foo=Table1()
#What should work, but doesn't
select(Table2.id).where(foo.bar(Table2.col1))

#Also tried
bar=foo.bar
select(Table2.id).where(func.bar(Table2.col1))
```

**Comment by CaselIT** [comment]
Hi,

I think that you are meant to call it as `Table1.bar(...)`

**Reply by DUOLabs333** [reply]
Yes, but that runs against all objects of `Table1`, not `foo` alone.

**Reply by CaselIT** [reply]
Sure, but what you are building there is a select that runs on a table, not on a single row.
Just can use other where clauses to limit the scope of the select.

I'm probably not understanding what you want to do here. Can you provide an example of method and what would you want to create a sql select to run in the db?

**Reply by DUOLabs333** [reply]
Ok, let's say
```
def bar(self, baz):
     return self.col1+baz.col1
```

I want `select(Table2.id).where(foo.bar(Table2)>10)` to select all rows of `Table2` such that `Table2.col1+foo.col1>10` (yes, I know that I can just copy the body, but I want to avoid that, as I want to avoid having to edit multiple places if I want to add a feature to `bar`).

**Reply by CaselIT** [reply]
If `foo.col1` is a number, I'm assuming it is since you are using an instance, just use a normal method
